cmake/FPrime.cmake

```diff
@@ -12,6 +12,11 @@ include(options)
 include(sanitizers) # Enable sanitizers if they are requested
 include(required)
 
+# Add project root's cmake folder to module path
+if (IS_DIRECTORY "${FPRIME_PROJECT_ROOT}/cmake")
```

Comment:
Could this cause an issue if projects using the new build structure have a folder called `cmake/` at the top level of their project? e.g. to store platform files or something

Reply:
This is to support exactly that use case

Reply:
When a project stores platform files, autocoders, and targets, these things are intended to be on the CMAKE_MODULE_PATH....which is why it is being added